Source/JavaScriptCore/jit/AssemblyHelpers.cpp

```diff
@@ -428,6 +429,17 @@ void AssemblyHelpers::emitLoadStructure(VM&, RegisterID source, RegisterID dest)
     emitNonNullDecodeZeroExtendedStructureID(dest, dest);
 }
 
+void AssemblyHelpers::emitEncodeStructure(RegisterID source, RegisterID dest)
+{
+#if ENABLE(STRUCTURE_ID_WITH_SHIFT)
+    urshift64(source, TrustedImm32(StructureID::encodeShiftAmount), dest);
+#elif CPU(ADDRESS64)
+    and32(TrustedImm32(static_cast<uint32_t>(StructureID::structureIDMask)), source, dest);
```

Comment:
This should be an and64 or we should static_assert that structureIDMask <= UINT32_MAX

Reply:
Updated